src/OpenLoco/src/GameSaveCompare.cpp

```diff
@@ -560,31 +560,94 @@ namespace OpenLoco::GameSaveCompare
 
         if (tileElements1.size() != tileElements2.size())
         {
-            Logging::info("The TileElements sizes are different. Will compare up to the smallest TileElements size.");
+            Logging::info("The TileElements sizes are different.");
             Logging::info("Size of TileElements1 = {}", tileElements1.size());
             Logging::info("Size of TileElements2 = {}", tileElements2.size());
         }
+
         int elementCount = 0;
-        for (auto iterator1 = smaller.begin(), iterator2 = larger.begin(); iterator1 != smaller.end();
-             ++iterator1, ++iterator2)
+        auto iterator1 = tileElements1.begin();
+        auto iterator2 = tileElements2.begin();
+        for (auto y = 0; y < 384; ++y)
         {
-            if (iterator2 != larger.end())
+            for (auto x = 0; x < 384; ++x)
             {
-                S5::TileElement tile1 = tileElements1.at(elementCount);
-                S5::TileElement tile2 = tileElements2.at(elementCount);
+                auto allElementsOnTile = [](auto& iter) {
```

Comment:
yeah i could have used a span but this isn't exactly an area we care about performance.